databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java

```diff
@@ -29,6 +30,43 @@
  * guessing
  */
 public class ApiClient {
+  static class Builder {
+    private DatabricksConfig config;
+    private Timer timer;
+    private Function<Void, Map<String, String>> authenticateFunc;
```

Comment:
Can we use Supplier for these?

Reply:
Suppliers look good for replacing the use of `Function` here, but some functions (`authenticate` as an example) are overloaded, and Supplier and the `::` syntax can't deal with those. So if we choose to use Supplier here, _some_ functions will still be stored under a `Function` variable, which kinda makes it even more complicated :/